src/main/java/io/supertokens/webserver/RequestStats.java

```diff
@@ -91,17 +92,24 @@ private void updateCounts(long currentSecond) {
         currentMinuteRequestCounts[(int) (currentSecond % 60)]++;
     }
 
-    public static RequestStats getInstance(Main main, AppIdentifier appIdentifier) {
+    public static RequestStats getInstance(Main main, AppIdentifier appIdentifier) throws TenantOrAppNotFoundException {
         try {
             return (RequestStats) main.getResourceDistributor()
                     .getResource(appIdentifier.getAsPublicTenantIdentifier(), RESOURCE_KEY);
         } catch (TenantOrAppNotFoundException e) {
+            if (Multitenancy.getTenantInfo(main, appIdentifier.getAsPublicTenantIdentifier()) == null) {
```

Comment:
add a comment explaining why this check is needed here but not in other places with similar code structure.